static/src/javascripts/projects/common/modules/identity/sign-in-gate/show-variants.js

```diff
@@ -0,0 +1,211 @@
+        if (articleBodyFirstChild) {
+            // container div to hold our "shadow" article dom while we create it
+            const shadowArticleBody = document.createElement('div');
+            // add the article body classes to the "shadow"
+            shadowArticleBody.className = articleBody.className;
+
+            // set the new article body to be first paragraph with transparent overlay, with the sign in gate component
+            shadowArticleBody.innerHTML = setTemplate({
+                child: articleBodyFirstChild,
+                guUrl,
+                signInUrl,
+                variant: abTest.variant,
+            });
+
+            // check if comment, and add comment/opinion bg colour
+            addOpinionBgColour({
+                element: shadowArticleBody,
+                target: '.signin-gate__first-paragraph-overlay',
+            });
+
+            // check page type/pillar to change text colour of the sign in gate
+            addPillarColour({
+                element: shadowArticleBody,
+                target: '.signin-gate__benefits--text',
+            });
+
+            // add click handler for the dismiss of the gate
+            addClickHandler({
+                element: shadowArticleBody,
+                target: '.js-signin-gate__dismiss',
+                abTest,
+                component,
+                value: 'dismiss',
+                callback: () => {
+                    // show the current body. Remove the shadow one
+                    articleBody.style.display = 'block';
+                    shadowArticleBody.remove();
+
+                    // Tell other things the article has been redisplayed
+                    mediator.emit('page:article:redisplayed');
+
+                    // user pref dismissed gate
+                    setUserDismissedGate({
+                        componentName,
+                        name: abTest.name,
+                        variant: abTest.variant,
+                    });
+                },
+            });
+
+            // add click handler for sign in button click
+            addClickHandler({
+                element: shadowArticleBody,
+                target: '.js-signin-gate__button',
+                abTest,
+                component,
+                value: 'signin_button',
+            });
+
+            // add click handler for the why sign in link
+            addClickHandler({
+                element: shadowArticleBody,
+                target: '.js-signin-gate__why',
+                abTest,
+                component,
+                value: 'why_sign_in',
+            });
+
+            // Hide the article Body. Append the shadow one.
+            articleBody.style.display = 'none';
+            if (articleBody.parentNode) {
+                articleBody.parentNode.appendChild(shadowArticleBody);
+            }
+        }
+    }
+};
+
+// 'show' method for the 'control' variant
+export const control: ({
```

Comment:
Can these be moved into their own files ? It would be better to have a single interface for the Sign-in component, and each variant for be an instance of a Sign-in component. Then it makes it easier to abstract more of the common functionality out , and also allow the creation of new variants to be simpler.